prime-router/metadata/fhir_mapping/hl7/secondary/Source.yml

```diff
@@ -1,8 +1,35 @@
-name:
+name_1:
   type: STRING
   valueOf: MSH.3
   expressionType: HL7Spec
 
+# For all other ID types
+name_2:
+  type: URL
+  condition: $idStr NOT_NULL && $systemStr NOT_EQUALS ISO && $systemStr NOT_EQUALS UUID && $systemStr NOT_EQUALS DNS && $systemStr NOT_EQUALS URI && $systemStr NOT_EQUALS CLIA
+  valueOf: $value
+  expressionType: HL7Spec
+  vars:
+    idStr: String, MSH.3.2
+    systemStr: String, MSH.3.3
+    value: MSH.3.1 + $dash + MSH.3.3 + $colon + MSH.3.2
+  constants:
```

Comment:
Note: Something to keep in mind that you cannot just inline these. E.g. you cannot do `value: MSH.3.1 + "-" + ...`

Reply:
This is worth putting somewhere in kdoc or in plain doc. It sounds like something that would easily get someone stuck for a while when updating configs

Reply:
Will create an issue to go into the Linux4Heatlh repo and submit a PR with documentation changes.

Reply:
We talked about it, and a ticket will be written to address these later, focusing on the library itself